salad/tests/util.py

```diff
@@ -17,3 +29,81 @@ def assert_with_negate(assertion, negator=None):
         assert not assertion
     else:
         assert assertion
+
+
+def assert_value(type_of_match, value, text, negate):
+    """
+    assert that a value equals or is contained in a text
+    the logic is reversed if negate is True
+    """
+    if type_of_match == 'is':
+        assert_equals_with_negate(value, text, negate)
+    elif type_of_match == 'contains':
+        assert_with_negate(value in text, negate)
+    else:
+        msg = ("type_of_match must be either 'is' or 'contains', but "
+               "you used '%s'." % (type_of_match, ))
+        raise NotImplementedError(msg)
+
+
+def store_with_case_option(key, value, upper_lower):
+    """
+    store an original / uppercase / lowercase value in world.stored_values
+    with the key 'key'
+    world.stored_values[key] will be overwritten!
+    no check if it already exists!
+    """
+    if not upper_lower:
+        world.stored_values[key] = value
+        return
+    if 'lower' in upper_lower:
+        world.stored_values[key] = value.lower()
+    elif 'upper' in upper_lower:
+        world.stored_values[key] = value.upper()
+    else:
+        msg = ("upper_lower must contain either 'upper' or 'lower', but "
+               "you used '%s' as input." % (upper_lower, ))
+        raise NotImplementedError(msg)
+
+
+def transform_for_upper_lower_comparison(stored, current, upper_lower):
```

Comment:
This whole pattern should be simplified to something like

"contains foo" - no transformation
"contains foo, case-insensitive"  - both the match and the key are `.lower()`-cased, and compared.  Upper should not exist, and if the default is case-sensitive, no need to explicitly set it.

+1 to changing the storage function to the above, and simplifying this function.